python/lsst/daf/butler/registry/interfaces/_database.py

```diff
@@ -1202,8 +1202,12 @@ def safeNotEqual(a: Any, b: Any) -> bool:
                 # ones that perform similar logic.
                 n, bad, result = check()
                 if n < 1:
-                    raise RuntimeError(
-                        "Necessary insertion in sync did not seem to affect table.  This is a bug."
+                    raise ConflictingDefinitionError(
+                        f"Attempted to ensure {row} exists by inserting it with ON CONFLICT IGNORE, "
+                        f"but a post-insert query on {keys} returned no results. "
+                        f"Insert was {'' if inserted else 'not '} reported as successful. "
```

Comment:
Space after `not` is not necessary. Or maybe space after closing curly brace needs to go?